### PR TITLE
Skeleton of automated setup support for MacOS.

### PR DESCRIPTION
**Pull request by hmschreck**

### CODE REVIEW

**Comment by mpdehaan** [comment]
Hi!

I thought I commented on this earlier but maybe it got lost.  This would be nice to have and thanks for the idea.  While Vespene is something I mostly develop on a Mac, I can see it being useful for a quick test drive.

I have typically been using Postgres.app but don't really care for it and don't mind suggesting the homebrew copy.

As most people using this would be using a developer setup, what do you think about having 6_services.sh exit out with a message?

I suspect we'd need to add some notes to setup.rst in the "docs/" folder as well.

**Comment by mpdehaan** [comment]
I actually need this myself for testing those setup scripts so going to merge this just now and look into that latter part.  Thank you!

**Comment by hmschreck** [comment]
@mpdehaan I just updated with notes in Setup.rst as well as an alert from the services.sh file that it will be set up in "developer mode".

**Comment by mpdehaan** [comment]
So far things are pretty good, one problem I have been finding out is that normally (because OS X) I'm launching the command as "sudo bash 2_database.py" and the brew services step that starts postgres is getting run as root, which is causing postgres to die, because it is REFUSING to run as root.

Did you experience this and how did you work around it?

**Comment by mpdehaan** [comment]
Minor, but I'm going to move that if clause to the top for readability - just a quick exit

**Comment by mpdehaan** [comment]
(and printing out the command line to run supervisor)

**Comment by mpdehaan** [comment]
Can you take a look at https://github.com/vespene-io/vespene/tree/osx_setup ?  This includes some minor changes but also my merge of the bind options for gunicorn.

I've moved the services stuff around to output the supervisor command but am still not *quite* sure of the best way to start postgresql in 2_database.sh if people are going to run the whole thing with sudo.

**Comment by hmschreck** [comment]
I didn't experience the postgres thing because I already have it set up on my machine.  I think explicitly setting sudo within the scripts is likely to be better.  Trading minor annoyance for it, but it's fine :)

**Comment by mpdehaan** [comment]
Yeah I am assuming I shouldn't sudo the initdb's on OS X but I'm not sure.

I mean, I'm going to be running postgresql as *me* which means the user in the pg_hba.conf might also need to be me?

**Comment by mpdehaan** [comment]
I'm going to work on making vespene run as a 'vespene' user for all non Mac platforms while I'm at it.

File: setup/2_database.sh [hunk 1]
```diff
@@ -38,9 +40,14 @@ host	all	all	0.0.0.0/0	md5
...
 
 echo "starting postgresql..."
+if [ "$OSTYPE" == "linux-gnu" ]; then
 # start the PostgreSQL service using systemd
-systemctl enable postgresql.service
-systemctl start postgresql.service
+    systemctl enable postgresql.service
+    systemctl start postgresql.service
+elif [ "$DISTRO" == "MacOS" ]; then
+    brew services start postgresql
```
Comment (4383) [inline setup/2_database.sh]: I'm not a Mac user so I don't have experince with homebrew but we don't need to enable the service to ensure us that we automaticaly run it at system start?

Like `systemctl enable postgresql.service` on linux

Comment (hmschreck) [reply setup/2_database.sh]: Brew is weird - when you use it to start a service, it has roughly equivalent effect to starting and enabling using systemd.

**Comment by mpdehaan** [comment]
Hey just to avoid duplicate work I have some things on master that aren't ready yet.  hold off for now, as everything is going to be a merge conflict.  When I get done it can definitely use some retesting on OS X, and I'm going to do some testing on clean Ubuntu/CentOS systems next.

**Comment by mpdehaan** [comment]
pushed a bit on the "new_setup" branch, seems good on Ubuntu after trying on OS X.

Next steps, trying on CentOS, changes as needed

Then retry on OS X, then push.

**Comment by mpdehaan** [comment]
Ok, merged in (your patches are included) - thoughts welcome.

Just to keep this from being in five places let's use this thread - https://talk.vespene.io/t/setup-overhaul-app-now-runs-as-vespene-on-linux-os-x-homebrew-setup-etc/35

**Pull request closed without merging**